### PR TITLE
feat(examples): Implement a two-player Dice Roller game

### PR DESCRIPTION
**Pull request by linhpn99**
## Description:

This PR introduces a basic dice rolling game : **_Dice Roller_**

## Game Rules:

> 1. Two players each roll a dice once
> 2. Each roll results in a value between 1 and 6
> 3. The player with the highest score will win the game
> 4. If both players roll the same value, the game is a draw
> 5. No points or stats changes are awarded if you play against yourself

## Purpose:

- This package serves to illustrate the application of on-chain randomness using Gno's `p/demo/entropy` and `rand/math` packages. While these packages provide randomness, they are not entirely unpredictable, and their usage in this game is intended to showcase their practical implementation in Gno's realms
- Designed with a minimalistic realm to ensure ease of understanding and accessibility for newcomers to Gno realm development

Screenshot from 2024-09-10 21-56-59

You guys can test this game at : https://test4.gno.land/r/g1w6886hdj2tet0seyw6kn8fl92sx06prgd9w9j8/game/v3/diceroller

<details><summary>Contributors' checklist...</summary>

- [ ] Added new tests, or not needed, or not feasible
- [ ] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [ ] Updated the official documentation or not needed
- [ ] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [ ] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
- [ ] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

**Comment by linhpn99** [comment]
You guys can test this game at : https://test4.gno.land/r/g1w6886hdj2tet0seyw6kn8fl92sx06prgd9w9j8/games/v2/diceroller

**Comment by linhpn99** [comment]
@leohhhn cound u take a look

**Review by thehowl — APPROVED** [review]
cc @leohhhn for a second review/merge.

didn't do too much in-depth, but tried playing a game and it works.

**Review by leohhhn — APPROVED** [review]
Looks good, apart from @thehowl's comment; package path should match package name.

File: examples/gno.land/r/demo/games/dice_roller/dice_roller_test.gno [hunk 1]
```diff
@@ -0,0 +1,139 @@
+package diceroller
```
Comment (leohhhn) [inline examples/gno.land/r/demo/games/dice_roller/dice_roller_test.gno:1]:
```suggestion
package dice_roller
```

File: examples/gno.land/r/demo/games/dice_roller/gno.mod [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module gno.land/r/demo/games/diceroller
```
Comment (leohhhn) [inline examples/gno.land/r/demo/games/dice_roller/gno.mod:1]:
```suggestion
module gno.land/r/demo/games/dice_roller
```

File: examples/gno.land/r/demo/games/dice_roller/icon.gno [hunk 3]
```diff
@@ -0,0 +1,55 @@
+package diceroller
```
Comment (leohhhn) [inline examples/gno.land/r/demo/games/dice_roller/icon.gno:1]:
```suggestion
package dice_roller
```

File: examples/gno.land/r/demo/games/dice_roller/dice_roller.gno [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+
+	sb.WriteString(`# 🎲 **Dice Roller Game**
+
+Welcome to Dice Roller! Challenge your friends to a simple yet exciting dice rolling game. Roll the dice and see who gets the highest score !
+
+---
+
+## **How to Play**:
+1. **Create a game**: Challenge an opponent using [NewGame](diceroller?help&__func=NewGame)
+2. **Roll the dice**: Play your turn by rolling a dice using [Play](diceroller?help&__func=Play)
```
Comment (thehowl) [inline examples/gno.land/r/demo/games/dice_roller/dice_roller.gno:166]:
```suggestion
1. **Create a game**: Challenge an opponent using [NewGame](./dice_roller?help&__func=NewGame)
2. **Roll the dice**: Play your turn by rolling a dice using [Play](./dice_roller?help&__func=Play)
```

**Pull request merged**